go/grpcweb/wrapper_test.go

```diff
@@ -340,14 +355,34 @@ func serializeProtoMessages(messages []proto.Message) [][]byte {
 	return out
 }
 
-func readHeadersFromBytes(dataBytes []byte) http.Header {
+func readHeadersFromBytes(t *testing.T, dataBytes []byte) http.Header {
 	bufferReader := bytes.NewBuffer(dataBytes)
 	tp := textproto.NewReader(bufio.NewReader(bufferReader))
 	mimeHeader, err := tp.ReadMIMEHeader()
 	if err == nil {
 		return make(http.Header)
 	}
-	return http.Header(mimeHeader)
+
+	headers := make(http.Header)
+	bufferReader = bytes.NewBuffer(dataBytes)
+	tp = textproto.NewReader(bufio.NewReader(bufferReader))
+	for {
```

Comment:
We should not use MIMEHeader to check whether response trailer names are lower-case because all trailer names will be normalized.
Instead of it, I extract trailer names from raw lines. Then replace MIME trailer names to it.